skyportal/plot.py

```diff
@@ -1380,7 +1381,9 @@ def spectroscopy_plot(obj_id, user, spec_id=None, width=600, device="browser"):
 
     obj = Obj.get_if_accessible_by(obj_id, user)
     spectra = (
-        Spectrum.query_records_accessible_by(user)
+        Spectrum.query_records_accessible_by(
+            user, options=[joinedload(Spectrum.annotations)]
```

Comment:
This is not right -- you'll need to use a subquery here, becase annotations on spectra can have different permissioning than their associated spectrum (access to annotations on spectra is determined by group membership, not only whether they can read the underlying spectrum).

Reply:
OK so query spectrum annotations (accessible by) and then filter on spectrum ID. Got it.